tests/transforms/test_split_non_commuting.py

```diff
@@ -13,105 +13,206 @@
+
+        spy = mocker.spy(qml.math, "concatenate")
+
+        assert len(split) == 1
+        assert all(isinstance(t, qml.tape.QuantumScript) for t in split)
+        assert fn([0.5]) == 0.5
+
+        # test transform on qscript
+        qs = qml.tape.QuantumScript(tape.operations, tape.measurements)
+        split, fn = split_non_commuting(qs)
+
+        assert len(split) == 1
+        assert all(isinstance(i_qs, qml.tape.QuantumScript) for i_qs in split)
+        assert fn([0.5]) == 0.5
+
+        spy.assert_not_called()
+
+    @pytest.mark.parametrize("meas_type", obs_meas_fn)
+    @pytest.mark.parametrize("obs_list, expected", [(obs_list_2, 2), (obs_list_3, 3)])
+    def test_non_commuting_group_right_number(self, meas_type, obs_list, expected):
+        """Test that the no. of tapes after splitting into commuting groups is of the right size"""
+
+        # create a queue with several measurements of same type but with differnent non-commuting
+        # observables
+        with qml.queuing.AnnotatedQueue() as q:
+            qml.PauliZ(0)
+            qml.Hadamard(0)
+            qml.CNOT((0, 1))
+            for obs in obs_list:
+                meas_type(op=obs)
+
+            # if MP type can accept wires, then add two extra measurements using wires and test no.
+            # of tapes after splitting commuting groups
+            if meas_type in wire_meas_fn:
+                meas_type(wires=[0])
+                meas_type(wires=[0, 1])
+
+        tape = qml.tape.QuantumScript.from_queue(q)
         split, _ = split_non_commuting(tape)
         assert len(split) == expected
 
         qs = qml.tape.QuantumScript(tape.operations, tape.measurements)
         split, _ = split_non_commuting(qs)
         assert len(split) == expected
 
+    @pytest.mark.parametrize("meas_type", obs_meas_fn)
     @pytest.mark.parametrize(
-        "tape,group_coeffs",
-        [(non_commuting_tape2, [[0, 2], [1, 3]]), (non_commuting_tape3, [[0, 3], [1, 4], [2, 5]])],
+        "obs_list, group_coeffs",
+        [(obs_list_2, [[1, 3], [0, 2, 4]]), (obs_list_3, [[0, 3], [1, 4], [2, 5]])],
     )
-    def test_non_commuting_group_right_reorder(self, tape, group_coeffs):
+    def test_non_commuting_group_right_reorder(self, meas_type, obs_list, group_coeffs):
         """Test that the output is of the correct order"""
-        split, fn = split_non_commuting(tape)
-        assert all(np.array(fn(group_coeffs)) == np.arange(len(split) * 2))
+        # create a queue with several measurements of same type but with differnent non-commuting
+        # observables
+        with qml.queuing.AnnotatedQueue() as q:
+            qml.PauliZ(0)
+            qml.Hadamard(0)
+            qml.CNOT((0, 1))
+            for obs in obs_list:
+                meas_type(op=obs)
+
+        tape = qml.tape.QuantumScript.from_queue(q)
+        _, fn = split_non_commuting(tape)
+        assert all(np.array(fn(group_coeffs)) == np.arange(len(tape.measurements)))
```

Comment:
For `obs_list_2` and `obs_list_3`, they are as many groups as measurements?

Reply:
No, there are 2 and 3 groups respectively. This here is to test the `reorder_fn` which ensures that the measurement results are returned in the correct order.